include/zephyr/bluetooth/mesh/brg_cfg_cli.h

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_MESH_BRG_CFG_CLI_H__
+#define ZEPHYR_INCLUDE_BLUETOOTH_MESH_BRG_CFG_CLI_H__
+
+#include <zephyr/bluetooth/mesh.h>
+#include <zephyr/bluetooth/mesh/brg_cfg.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup bt_mesh_brg_cfg_cli Bridge Configuration Client Model
+ * @ingroup bt_mesh
+ * @{
+ *  @brief API for the Bluetooth Mesh Bridge Configuration Client model
+ */
+
+struct bt_mesh_brg_cfg_cli;
+
+/**
+ *  @brief Bridge Configuration Client model Composition Data entry.
+ *
+ *  @param _cli Pointer to a @ref bt_mesh_brg_cfg_cli instance.
+ */
+#define BT_MESH_MODEL_BRG_CFG_CLI(_cli)                                                            \
+	BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_BRG_CFG_CLI, _bt_mesh_brg_cfg_cli_op, NULL, _cli,        \
+			 &_bt_mesh_brg_cfg_cli_cb)
+
+/** Mesh Bridge Configuration Client Status messages callback */
+struct bt_mesh_brg_cfg_cli_cb {
+	/** @brief Optional callback for Subnet Bridge Status message.
+	 *
+	 *  Handles received Subnet Bridge Status messages from a Bridge
+	 *  Configuration Server.
+
+	 *  @param cli      Bridge Configuration Client context.
+	 *  @param addr     Address of the sender.
+	 *  @param status   Status received from the server.
+	 */
+	void (*subnet_bridge_status)(struct bt_mesh_brg_cfg_cli *cli, uint16_t addr,
```

Comment:
I think we should move away from individual msg callbacks to a single callback that dispatches events (or even opcodes) to let application parse whatever messages it wants. The MDF has 20+ status messages and this will become pretty long list.

Reply:
Will not be strange, that we suddenly for this model change the way we do it. For cfg_cli_cb we have 22 callbacks, so the amount here is not a good reason for changing this. So I think there should be another good reason for doing this another way. It is easier for the users, that there at least, is a consistent way of doing this. That way the users do not need to do this in another way for subnet bridge. That is why I want to keep it as it is.

Reply:
Agree with @Balaklaka. Architecture design should follow a similar pattern. It will look very not professional if models have different architecture approach.